src/Symfony/Component/Security/Http/Attribute/IsGranted.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Attribute;
+
+/**
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+#[\Attribute(\Attribute::IS_REPEATABLE | \Attribute::TARGET_CLASS | \Attribute::TARGET_METHOD | \Attribute::TARGET_FUNCTION)]
+class IsGranted
```

Comment:
should probably be final, because the usage of `$event->getAttributes()[IsGranted::class]` means that the listener does not support subclasses of that attribute.